### ISSUE TITLE
Add Gitlab Support

### ISSUE THREAD

**Issue by bborysenko** [ansible/galaxy-issues#96]
Would be great if we can import roles from Gitlab too. 

We are going to move some part of our roles to Gitlab and test them with Gitlab CI. But we still have to maintain read-only repo mirror on Github, to be able upload roles to Galaxy

**Comment by chouseknecht** [comment]
This was suggested by Peter Spygada as well.

+1

**Comment by resmo** [comment]
+1

**Comment by ksatirli** [comment]
I'd love to see support for more (hosted / self-hosted) Git servers happen. Not just Gitlab, but also Github Enterprise and of course: Bitbucket.

**Comment by GR360RY** [comment]
+1

**Comment by S1y** [comment]
what status ?

**Comment by marcowork** [comment]
+1

**Comment by jbarascut** [comment]
How long will it take to get this feature fixed? 
I'm trying to use a private ansible-galaxy and I would like to help you.

My principal problem is that my network have no connection to internet (local bitbucket only).

**Comment by ReSearchITEng** [comment]
Is there a document to try configuring it in intranet? 
Can anyone share the issues faced while trying to point it to gitlab?

**Comment by rainerduffner** [comment]
What is the status of this enhancement?
We have a private gitlab instance that we would want to use.

**Comment by marcowork** [comment]
I asked Redhat the other day if and when they are going to implement this feature. The awnser was a bit disappointing. 

As this is a project and not a product, redhat is not going to do anything. They expext the communicaty to create this feature.

As Gitlab is just another SCM I would actually say, lets fork this and creat a galaxy that fits on every SCM. Also bitbucked, SVN and others.....

**Comment by ReSearchITEng** [comment]
@marcowork For anyone to start this would require some clarifications where the current code resides. Does anyone know where is the code that needs to be improved ?

**Comment by bcoca** [comment]
@marcowork not sure why that requires a fork, just submit PR against https://github.com/ansible/galaxy

**Comment by marcowork** [comment]
@bcoca you are right. It should not be a fork. But a PR is a bit to much to rewrite the whole project. But I can give it a try

**Comment by bcoca** [comment]
you don't need to do the whole thing in 1 PR
